packages/data/addon/serializers/metadata/bard.ts

```diff
@@ -283,23 +335,44 @@ export default class BardMetadataSerializer extends EmberObject implements NaviM
     parameters: RawColumnFunctionArguments,
     dataSourceName: string
   ): FunctionParameterMetadataPayload[] {
-    return Object.keys(parameters).map(param => {
-      const { type, defaultValue, values, dimensionName, description } = parameters[param];
+    return Object.keys(parameters).map(paramName => {
+      const param = parameters[paramName];
+      const { defaultValue, description } = param;
 
       const normalized: FunctionParameterMetadataPayload = {
-        id: param,
-        name: param,
+        id: paramName,
+        name: paramName,
         description,
         type: 'ref', // It will always be ref for our case because all our parameters have their valid values defined in a dimension or enum
-        expression: type === 'dimension' ? `dimension:${dimensionName}` : INTRINSIC_VALUE_EXPRESSION,
-        _localValues: values,
+        expression: param.type === 'dimension' ? `dimension:${param.dimensionName}` : INTRINSIC_VALUE_EXPRESSION,
+        _localValues: param.type === 'enum' ? param.values : undefined,
         source: dataSourceName,
         defaultValue
       };
       return normalized;
     });
   }
 
+  /**
+   * Defaults the supported grains to the day grain of the table in utc timezone
+   * @param table - raw table
+   * @param dimensions - raw dimensions
+   * @param dataSourceName - data source name
+   */
+  private normalizeTimeDimensions(
+    table: RawTablePayload,
+    dimensions: RawDimensionPayload[],
+    dataSourceName: string
+  ): TimeDimensionMetadataPayload[] {
+    return this.normalizeDimensions(dimensions, dataSourceName)
+      .filter(d => d.valueType === 'date')
```

Comment:
Do we have a test that excutes this?

Reply:
Yep, we see it from this timeDimension
https://github.com/yahoo/navi/blob/c96e5c5404e4fd9d1cc097d4220d15b454809161/packages/data/tests/unit/serializers/metadata/bard-test.ts#L407-L414